### PR TITLE
fix(snowflake): ensure that quoting matches snowflake behavior, not sqlalchemy

### PR DESCRIPTION
**Pull request by cpcloud**
This PR addresses broken snowflake-sqlalchemy behavior that appears to have been stuck in open source limbo
for over three years. Numerous issues related to the snowflake-sqlalchemy dialect's broken quoting behavior exist:

- https://github.com/snowflakedb/snowflake-sqlalchemy/issues/157
- https://github.com/snowflakedb/snowflake-sqlalchemy/issues/276
- https://github.com/snowflakedb/snowflake-sqlalchemy/issues/388

with no obvious traction or clear effort towards addressing them.

So, in this PR I just disable snowflake-sqlalchemy's broken case-guessing and
let snowflake do what it will, and return what it will.

### CODE REVIEW

**Comment by gforsyth** [comment]
Hmm, should we retarget this at #5736 to avoid the mismatch in the (nearly) standardized keyword arguments?

**Comment by cpcloud** [comment]
I'm okay rebasing once this PR is merged.

**Review by gforsyth — APPROVED** [review]
:shipit:

File: ibis/backends/base/sql/alchemy/__init__.py [hunk 1]
```diff
@@ -413,12 +416,14 @@ def _log(self, sql):
             util.log(query_str)
 
     def _get_sqla_table(
-        self, name: str, schema: str | None = None, autoload: bool = True, **kwargs: Any
+        self, name: str, schema: str | None = None, autoload: bool = True, **_: Any
```
Comment (jgehrcke) [inline ibis/backends/base/sql/alchemy/__init__.py:419]: I have always loved this naming convention :heart:

**Comment by cpcloud** [comment]
Merging.

**Pull request merged**